src/stcal/alignment/util.py

```diff
@@ -20,16 +20,19 @@
 log = logging.getLogger(__name__)
 log.setLevel(logging.DEBUG)
 
-
 __all__ = [
     "compute_scale",
     "compute_fiducial",
     "calc_rotation_matrix",
+    "update_s_region_imaging",
+    "update_s_region_keyword",
     "wcs_from_footprints",
     "reproject",
+    "SupportsDataWithWcs"
 ]
 
 
+@runtime_checkable
 class SupportsDataWithWcs(Protocol):
     _asdf: AsdfFile
```

Comment:
In `util.py`, `ref_model` is used in only one way: `refmodel.meta.wcs`. Here it can definitely be replaced by `ref_wcs` so that we do not need an interface to data models.

Reply:
This is mostly true; it's used only for `model.meta.wcs` and `model.meta.wcsinfo`.  Plus there is a singular instance where `model.meta.coordinates.reference_frame` is checked, in the function `wcsinfo_from_model()`.  I agree with you that these instances can all be changed to take `wcs`, `wcsinfo`, and `referece_frame` directly as inputs.

There are some consequences to doing that.  The call structure of the `update_s_region_imaging`, `update_s_region_keyword`, and `wcs_from_footprints` functions have to change, and these are all used directly in the JWST repository.  These changes maybe slightly decrease ease-of-use: e.g. for `wcs_from_footprints()` there are now three required arguments instead of just one.  But I think that's okay.

I'm still a bit torn about whether something like this protocol should be implemented for stcal/tweakreg.  At present, that module uses `model.meta.wcs`, `model.meta.wcsinfo`, `model.meta.group_id`, and `model.meta.observation.date`.  However, after doing the refactor of `util.py` it seems a lot more tractable to remove the dependencies on datamodels entirely... I'm going to see what that looks like now.

Reply:
How do the new changes look to you @mcara ?  The corresponding jwst PR incorporates the changes too and is found [here](https://github.com/spacetelescope/jwst/pull/8624).